src/main/java/de/codingair/tradesystem/utils/database/migrations/mysql/MysqlMigrations.java

```diff
@@ -0,0 +1,92 @@
+package de.codingair.tradesystem.utils.database.migrations.mysql;
+
+import de.codingair.tradesystem.utils.database.migrations.Migration;
+import de.codingair.tradesystem.utils.database.migrations.SqlMigrations;
+import org.bukkit.Bukkit;
+
+import javax.sql.DataSource;
+import java.sql.*;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MysqlMigrations implements SqlMigrations {
+
+    private static MysqlMigrations instance;
+    // Define all migrations in this list.
+    private static List<Migration> migrations = Arrays.asList(new CreateTradeLogTableMigration());
+    private final DataSource datasource;
+
+    private MysqlMigrations() {
+        datasource = MySQLConnection.getDatasource();
+    }
+
+    public static MysqlMigrations getInstance() {
+        if (instance == null) {
+            instance = new MysqlMigrations();
+        }
+        return instance;
+    }
+
+    @Override
+    public void createMigrationTable() {
+        try (Connection connect = datasource.getConnection()) {
+            Bukkit.getLogger().info("Creating migration table");
+
+            // SQL statement for creating a new table
+            String sql = "CREATE TABLE IF NOT EXISTS migrations (\n"
+                    + "	id BIGINT PRIMARY KEY AUTO_INCREMENT,\n"
+                    + "	version integer NOT NULL\n"
+                    + ");";
+            Statement stmt = connect.createStatement();
```

Comment:
Statement won't get closed. You can put that into the try brackets as well.

Reply:
done